MyCryptoHQ/MyCrypto
# Rsk Name Service integration

Closes #2292

### Description

The aim of this PR is to add the necessary code to integrate the RSK Name Service (RNS). As of now it works the same way as ENS does, so the changes are mostly to differentiate between networks and top level domains. The core logic is not affected except for the mentioned distinction.

Currently only the `rsk` top level is available through an auction based registrar.

### Changes

* Add code to get/select the corresponding main TLD (`eth` or `rsk`), other TLDs (`test`, `reverse`) as well as contract addresses based on the chain id
* Use the "main TLD" instead of considering only names finished in ".eth" when checking wether to resolve a name or not 
* Add/Modify ENS address validation functions to consider cases other than the ENS top level domains
* Add/Modify ENS/RNS related translations
* Modify tests where necessary
* EDIT: Add ABIs and addresses to the RSK network contracts

### Steps to Test

1. Select RSK Mainnet network
2. Go to /account/send
3. Type "martin.rsk" in the To field
4. It should resolve to address `0xfb530616391cb526387bad651594bc21a77d3dfe`

### Screenshots

![mycrypto_rns](https://user-images.githubusercontent.com/11276821/49522435-6cd4ed80-f886-11e8-9a87-42646eec92c9.png)
